### PR TITLE
Promote some experimental features to stable for release 2.9

### PR DESCRIPTION
**Pull request by flxbk**
#### What this PR does

This promotes the following features/options from experimental to stable:
- `-query-frontend.query-sharding-max-regexp-size-bytes`
- `-query-scheduler.max-used-instances`
- `-(alertmanager|blocks|ruler)-storage.storage-prefix`
- `-compactor.first-level-compaction-wait-period`
- `-usage-stats.*`
- `-query-frontend.query-sharding-target-series-per-shard`

It also adds a non-default value of `2500` for the `-query-frontend.query-sharding-target-series-per-shard` flag in Jsonnet and Helm. We've obtained good results from this setting internally at Grafana Labs, but enabling it by default would imply enabling the results cache by default, which we don't want to do.

Individual commits are self-contained to make review easier.

#### Which issue(s) this PR fixes or relates to

Related to #5112

#### Checklist

- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/mimir/modules_test.go [hunk 1]
```diff
@@ -231,6 +231,9 @@ func TestMultiKVSetup(t *testing.T) {
 			// Must be set, otherwise MultiKV config provider will not be set.
 			cfg.RuntimeConfig.LoadPath = []string{filepath.Join(dir, "config.yaml")}
 
+			// Cardinality estimation needs a cache backend if enabled, disable it for this test.
+			cfg.Frontend.QueryMiddleware.TargetSeriesPerShard = 0
```
Comment (flxbk) [inline pkg/mimir/modules_test.go:235]: I'm not sure this is the best way to fix this test.

File: pkg/compactor/compactor_test.go [hunk 2]
```diff
@@ -1656,6 +1656,9 @@ func prepareConfig(t *testing.T) Config {
 	compactorCfg.ShardingRing.Common.KVStore.Mock = inmem
 	compactorCfg.ShardingRing.Common.InstanceAddr = "localhost"
 
+	// The new default is 25m, but tests rely on the previous value of 0s
+	compactorCfg.CompactionWaitPeriod = 0
```
Comment (flxbk) [inline pkg/compactor/compactor_test.go:1660]: I didn't dive into the details of this test, not sure if we should make it work with the new default instead.

File: integration/e2emimir/services.go [hunk 3]
```diff
@@ -189,8 +189,9 @@ func NewCompactor(name string, consulAddress string, flags map[string]string, op
 			"-compactor.ring.store":           "consul",
 			"-compactor.ring.consul.hostname": consulAddress,
 			// Startup quickly.
-			"-compactor.ring.wait-stability-min-duration": "0",
-			"-compactor.ring.wait-stability-max-duration": "0",
+			"-compactor.ring.wait-stability-min-duration":   "0",
+			"-compactor.ring.wait-stability-max-duration":   "0",
+			"-compactor.first-level-compaction-wait-period": "0s",
```
Comment (flxbk) [inline integration/e2emimir/services.go:194]: This is needed to unbreak `TestCompactBlocksContainingNativeHistograms`, which apparently expects this value to be lower than the new default.

File: pkg/compactor/compactor_test.go (hunk 2 above)
Comment (krajorama) [reply pkg/compactor/compactor_test.go:1660]: I don't think it affects the compaction algorithm itself so I think this is ok. I assume the wait period is tested separately ? Anyway, should be out of scope for this PR

File: integration/e2emimir/services.go (hunk 3 above)
Comment (krajorama) [reply integration/e2emimir/services.go:194]: I think this is fine to not slow down the test.

File: CHANGELOG.md
Comment (krajorama) [inline CHANGELOG.md:1]: Let's mention the jsonnet query sharding target series per shard new default in the jsonnet section.

Similarly for helm but in its own changelog file.

Comment (flxbk) [reply CHANGELOG.md:1]: 👍  5bc71d66f

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 4]
```diff
@@ -28,6 +28,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 
 ## main / unreleased
 
+* [CHANGE] Query-frontend: enable cardinality estimation for query sharding by default if `results-cache.enabled` is true. #5128
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:31]:
```suggestion
* [CHANGE] Query-frontend: enable cardinality estimation via `frontend.query_sharding_target_series_per_shard` in the Mimir configuration  for query sharding by default if `results-cache.enabled` is true. #5128
```

**Review by krajorama — APPROVED** [review]
Approved with one suggestion

File: pkg/compactor/compactor_test.go (hunk 2 above)
Comment (pracucci) [reply pkg/compactor/compactor_test.go:1660]: It's fine to set it to 0 here.

**Pull request merged**